File: 38. Count and Say/solution.go
func countAndSay(n int) string {
	if n == 1 {
		return "1"
	}

	// get last state string
	lastStr := countAndSay(n - 1)
	count := 1
	temp := lastStr[0]
    var buf bytes.Buffer

	for i := 1; i < len(lastStr); i++ {
		if lastStr[i] == temp {
			count++
		} else {
			buf.WriteString(strconv.Itoa(count))
			buf.WriteString(string(temp))
			// reset to initial status
			count = 1
			temp = lastStr[i]
		}
	}

	buf.WriteString(strconv.Itoa(count))
	buf.WriteString(string(temp))
	return buf.String()
}

File: 38. Count and Say/solution.py
class Solution:
    def countAndSay(self, n: int) -> str:
        s = '1'
        for _ in range(n - 1):
            cur, temp, cnt = s[0], '', 0
            for d in s:
                if cur == d:
                    cnt += 1
                else:
                    temp += str(cnt) + cur
                    cur = d
                    cnt = 1
            temp += str(cnt) + cur
            s = temp
        return s